recipes/scikit-learn-intelex/meta.yaml

```diff
@@ -0,0 +1,64 @@
+{% set name = "scikit-learn-intelex" %}
+{% set version = "2021.2.1" %}
+{% set buildnumber = 0 %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/intel/scikit-learn-intelex/archive/{{ version }}.tar.gz
+  sha256: 5cfd513245e686d5503ae6ef2d5a17dbf5d0ca41d14b2322a492e1f816ece251
+
+build:
+  skip: true  # [osx]
+  number: {{ buildnumber }}
+  include_recipe: false
+  script: "{{ PYTHON }} setup_sklearnex.py install --single-version-externally-managed --record=record.txt"
+
+requirements:
+  host:
+    - python
+    - setuptools
+  run:
+    - python
+    - daal4py ==2020.3.1
+
+test:
+  requires:
+    - pandas
+    - scikit-learn
```

Comment:
No as package itself will support any installation of scikit that user have - from different package manager, standalone versions  etc.  So we wouldn't lie to define explicit dependency for package.

Reply:
While that's a noble idea, at conda-forge we add all dependencies to `run`. (By your logic, we don't even need to specify `daal4py` because it supports any installation of `daal4py` that the user may have.)

Reply:
>  In case we will define requirement we will be forcing users to get sci-kit-learn from our requirement definition.

How is that a bad thing?

> While we can patch any scikit-learn version and get it optimized.

According to docs, we have,

The latest release of Intel(R) Extension for Scikit-learn 2021.2 supports scikit-learn 0.21.X, 0.22.X, 0.23.X and 0.24.X.

Reply:
One of cases would be if user have other non-conda based scikit-learn installation.  So having dependency installed will also cause scikit to be downloaded from conda. 
So from this perspective it's more of an optional dependency.  Might be it can be added as run_constrained dependency that will use scikit >=0.21

Reply:
If we go down that path, `scipy` doesn't really need `numpy` then, because the user can have a non-conda based `numpy` package.

>  Might be it can be added as run_constrained dependency that will use scikit >=0.21

How do we ensure that the user has a non-conda based scikit-learn installation? How do we ensure that the non-conda scikit-learn is >=0.21?

Reply:
I think the main difference is that in case of scipy you are installing it and expecting to work out of box.
While here the expectation that this is really extension to scikit packge and you have to use it in your code.  Use of dependency on scikit would mean for users that they are good to go, but reality is that scikit-learn-intelex will not be applied.  So it's extension for scikit-learn vs scikit-learn with intel extension. 

Package validates scikit-learn version prior to patching. And from package perspective it's doesn't matter how scikit package was received.

Reply:
> I think the main difference is that in case of scipy you are installing it and expecting to work out of box.

I don't understand this. A person using `scikit-learn-intelex` shouldn't expect it to work out of the box?

Reply:
Thanks for discussion =)  Will move to run dependency